ide/projectapi/src/org/netbeans/spi/project/SingleMethod.java

```diff
@@ -64,6 +81,15 @@ public FileObject getFile() {
         return file;
     }
 
+    /**
+     * Returns the binary class name of the class containing the method.
+     
+     * @return Class name held by this object
+     */
+    public Optional<String> getBinaryClassName() {
```

Comment:
I prefer plain `String` that can be `null`. There is no `Optional` in this API yet, and there is no real reason to start with it as far as I can tell.

This is an opinion.